app/create-pr.js

```diff
@@ -0,0 +1,171 @@
+/** source for cloud func, not auto-deployed, copy/paste */
+
+const functions = require("@google-cloud/functions-framework");
+const { Octokit } = require("octokit");
+
+/**
+ * github fine-grained personal access token
+ * single repo
+ * read/write permissions for contents and pull requests
+ */
+const auth = process.env.GITHUB_API_KEY;
+const owner = "vincerubinetti";
+const repo = "captions";
+
+/** endpoint */
+functions.http("create-pr", async (request, response) => {
+  response.set("Access-Control-Allow-Origin", "*");
+  response.set("Access-Control-Allow-Methods", "POST");
+  response.set("Access-Control-Allow-Headers", "Accept, Content-Type");
+
+  if (request.method == "OPTIONS") return response.status(200).send();
+
+  try {
+    const pr = await createPr(request.body);
+    return response.status(200).json(pr);
+  } catch (error) {
+    return response.status(400).send(error.message);
+  }
+});
+
+/** create/update branch, file, pr */
+async function createPr(params, debug = false) {
+  /** auth with github */
+  const octokit = new Octokit({ auth });
+
+  /** get params */
+  const { branch, title, body, files = [] } = params || {};
+
+  /** check for missing params */
+  const missing = [];
+  if (!branch) missing.push("branch");
+  if (!title) missing.push("title");
+  if (!body) missing.push("body");
+  if (!files.length) missing.push("files");
+  if (missing.length) throw Error(`Insufficient ${missing.join(", ")}`);
+
+  /** get main branch */
+  let main;
+  try {
+    main = (
+      await octokit.rest.git.getRef({
+        owner,
+        repo,
+        ref: "heads/main",
+      })
+    ).data.object.sha;
+  } catch (error) {
+    if (debug) console.warn(error);
+    throw Error(`Couldn't get main branch: ${error?.response?.data?.message}`);
+  }
+
+  /** get existing branch */
```

Comment:
As written, this actually would let someone add to an existing pr. I think if someone were to do that, it would just overwrite the previous files, wholesale, with their own. Wouldn't make the OP of the PR happy, but the commit could still be reverted.